cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactStore.java

```diff
@@ -639,7 +640,7 @@ public boolean apply(PluginClass input) {
                 }
               }
             }
-            PluginKey pluginKey = new PluginKey(parentArtifactRange.getNamespace().toId(),
+            PluginKey pluginKey = new PluginKey(Id.Namespace.from(parentArtifactRange.getNamespace()),
```

Comment:
can we modify PluginKey to take the entity instead of using the deprecated classes